src/Symfony/Component/RateLimiter/CompoundLimiter.php

```diff
@@ -25,17 +25,22 @@ final class CompoundLimiter implements LimiterInterface
      */
     public function __construct(array $limiters)
     {
+        if ([] === $limiters) {
+            throw new LogicException("Limiters shouldn't be empty. Please specify at least one.");
+        }
         $this->limiters = $limiters;
     }
 
-    public function consume(int $tokens = 1): bool
+    public function consume(int $tokens = 1): LimitInterface
     {
-        $allow = true;
         foreach ($this->limiters as $limiter) {
-            $allow = $limiter->consume($tokens) && $allow;
+            $limit = $limiter->consume($tokens);
+            if (0 === $limit->getRemainingTokens()) {
+                return $limit;
+            }
         }
 
-        return $allow;
+        throw new LogicException("A limit object should always be returned from precedent foreach function.");
```

Comment:
It shouldn't. You're only returning now on the case that a limiter reached its limit, not when all limiter's pass.

You have to do something like:

```php
$minimalLimit = null;
foreach ($this->limiters as $limiter) {
    $limit = $limiter->consume($tokens);

    if (0 === $limit->getRemainingTokens()) {
        return $limit;
    }

    if (null === $minimalLimit || $limit->getRemainingTokens() < $minimalLimit->getRemainingTokens()) {
        $minimalLimit = $limit;
    }
}

return $minimalLimit;
```